sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -89,6 +89,73 @@ class CollationStringExpressionsSuite
     testRepeat("UNICODE_CI", 3, "abc", 2)
   }
 
+  test("Levenshtein expressions with collation") {
+    def prepareLevenshtein(
+        left: String,
+        right: String,
+        collation: String): Levenshtein = {
+      val collationId = CollationFactory.collationNameToId(collation)
+      val leftLit = Literal.create(left, StringType(collationId))
+      val rightLit = Literal.create(right, StringType(collationId))
+      Levenshtein(leftLit, rightLit)
+    }
+
+    Seq(
+      // scalastyle:off nonascii
+      CollationTestCase("", "", "UTF8_BINARY", 0),
+      CollationTestCase("", "something", "UTF8_BINARY", 9),
+      CollationTestCase("a", "a", "UTF8_BINARY", 0),
+      CollationTestCase("a", "A", "UTF8_BINARY", 1),
+      CollationTestCase("a", "a", "UTF8_BINARY_LCASE", 0),
+      CollationTestCase("a", "A", "UTF8_BINARY_LCASE", 0),
+      CollationTestCase("bd", "ABc", "UTF8_BINARY_LCASE", 2),
+      CollationTestCase("Xü", "Ü", "UTF8_BINARY_LCASE", 1),
+      CollationTestCase("Xũ", "Üx", "UTF8_BINARY_LCASE", 2),
+      CollationTestCase("", "something", "UTF8_BINARY_LCASE", 9),
+      CollationTestCase("sOmeThINg", "SOMETHING", "UTF8_BINARY_LCASE", 0),
+      CollationTestCase("sOmeThINg", "SOMETHING", "UNICODE", 5),
+      CollationTestCase("sOmeThINg", "SOMETHING", "UNICODE_CI", 0)
+      // scalastyle:on nonascii
+    ).foreach(c => checkEvaluation(prepareLevenshtein(c.s1, c.s2, c.collation), c.expectedResult))
+  }
+
```

Comment:
@cloud-fan Should we also add a test to CollationSuite Implicit casting to check if casting actually works for this expression?

Reply:
my thoughts are: yes

@dbatomic and myself will be finishing some refactoring PRs soon, and will likely request all PRs to comply with some new design before further reivew

Reply:
Added the tests to CollationSuite, please check.

Reply:
Yea we need both unit tests and end-to-end tests